src/classes/ATTD_CourseConnectionContact_TDTM.cls

```diff
@@ -0,0 +1,93 @@
+    * @param listNew the list of Attendance Events from trigger new. 
+    * @param listOld the list of Attendance Events from trigger old. 
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.). 
+    * @param objResult the describe for Attendance Events 
+    * @return dmlWrapper.  
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
+    TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+
+        DmlWrapper dmlWrapper = new DmlWrapper();
+
+        boolean recursionFlag = TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.ATTD_CourseConnectionContact_TDTM);
+
+        Map<ID, SObject> newmap = new Map<ID, SObject>(newlist);
+        
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert
+                || triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+                    
+            if (!recursionFlag) {
+                for (Attendance_Event__c attd : [SELECT Id,
+                                                        Contact__c,
+                                                        Course_Connection__c,
+                                                        Course_Connection__r.Contact__c
+                                                FROM Attendance_Event__c 
+                                                WHERE Id IN :newlist]) {
+                    if (attd.Course_Connection__c != null
+                        && attd.Course_Connection__r.Contact__c != null) {
+                        if (attd.Contact__c != null) {
+                            if (attd.Course_Connection__r.Contact__c != attd.Contact__c) {
+                                //If contact not null and course connection contact does not match contact on attendance event
+                                //Throw an error and guide users to correct their data
+                                Attendance_Event__c attdInContext = (Attendance_Event__c)newmap.get(attd.ID);
+                                attdInContext.Contact__c.addError(Label.AttendanceCourseConnectionContactMismatch);
+                            }
+                        } else {
+                            //Auto-populdate contact lookup field on attendance event
+                            attd.Contact__c = attd.Course_Connection__r.Contact__c;
+                            dmlWrapper.objectsToUpdate.add(attd);
+                        } 
+
+                    }
+                }
+            }     
+        }
+        
+        TDTM_TriggerHandler.processDML(dmlWrapper, true);
+        
+        //Reset recursion flag
+        dmlWrapper = null;
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ATTD_CourseConnectionContact_TDTM, false);
```

Comment:
We need to set it to true before line 85 or it will never be true, I think?

Reply:
Updated.